src/init.cpp

```diff
@@ -1976,9 +1974,17 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     connOptions.nLocalServices      = nLocalServices;
     connOptions.nMaxConnections     = nMaxConnections;
-    connOptions.nSendBufferMaxSize  = 1000*GetArg("-maxsendbuffer", 1*1000);
-    connOptions.nReceiveFloodSize   = 1000*GetArg("-maxreceivebuffer", 5*1000);
-
+    int64_t msb_unbound = GetArg("-maxsendbuffer", 1*1000);
+    if (msb_unbound > 0 && msb_unbound <= std::numeric_limits<unsigned int>::max())
+        connOptions.nSendBufferMaxSize = 1000 * static_cast<unsigned int>(msb_unbound); // from bytes to kilobytes
```

Comment:
Two things here (and also for line `1984`):
- given that we multiply it by `1000`, I believe we should check that `msg_unbound <= std::numeric_limits<unsigned int>::max() / 1000`, otherwise it would overflow
- I believe the comment is misleading. Shouldn't it be the other way round, `kilobytes to bytes`?